lib/src/presentation/bloc/card_bloc.dart

```diff
@@ -0,0 +1,100 @@
+import 'dart:async';
+
+import '../../core/util/card_constants.dart';
+import '../../data/model/card.dart';
+import '../../domain/repository/repository_card.dart';
+import 'interfaces/i_card_bloc.dart';
+
+class CardBloc extends ICardBloc {
+  final RepositoryCard repositoryCard;
+  final defaultEndpoint = kRacesDemonEndPointApi;
+  late int firstCardOfPage = cardBlocStartingPositionOfShowedCards;
+  late int lastCardOfPage = kCardBlocNumberOfCardsShowedPerPage;
+  int currentPage = kCardBlocStartingCurrentPage;
+  late List<HearthstoneCard> allCards = [];
+  int numberOfPagesCardList = kDefaultNumberOfPagesCardList;
+  bool isLoading = false;
+
+  CardBloc({required this.repositoryCard});
+
+  final StreamController<List<HearthstoneCard>> _streamController =
+      StreamController<List<HearthstoneCard>>.broadcast();
+
+  @override
+  void dispose() {
+    _streamController.close();
+  }
+
+  @override
+  Future<void> initialize() async {
+    await sinkCardList(
+      defaultEndpoint,
+    );
+  }
+
+  @override
+  Stream<List<HearthstoneCard>> get cardListStream => _streamController.stream;
+
+  @override
+  Future<void> sinkCardList(String endpoint) async {
+    isLoading = true;
+    currentPage = kCardBlocStartingCurrentPage;
+    await fillAllCardsList(
+      endpoint,
+    );
+    isLoading = false;
+    numberOfPagesCardList =
+        (allCards.length / kCardBlocNumberOfCardsShowedPerPage).ceil();
+    sinkCurrentCardList();
+  }
+
+  Future<void> fillAllCardsList(String endpoint) async {
+    allCards = await repositoryCard.addCardList(
+      endpoint,
+    );
+  }
+
+  Future<void> sinkCurrentCardList() async {
+    List<HearthstoneCard> tenCards = <HearthstoneCard>[];
+    firstCardOfPage = currentPage * kCardBlocNumberOfCardsShowedPerPage;
+    lastCardOfPage = firstCardOfPage + kCardBlocNumberOfCardsShowedPerPage;
+    lastCardOfPage < allCards.length
+        ? tenCards = allCards
+            .getRange(
+              firstCardOfPage,
+              lastCardOfPage,
+            )
+            .toList()
+        : tenCards = allCards
+            .getRange(
+              firstCardOfPage,
+              allCards.length,
+            )
+            .toList();
+
```

Comment:
You can create a List<Page>,  you can create a new model (Page) that allows you to have, the page and a list of HeartsrhoneCards for that page.

Reply:
Done.